XLS-0056d-batch/README.md

```diff
@@ -0,0 +1,503 @@
+
+#### 2.5.3. `Signers`
+
+This field is included if the account is signing with multi-sign (as opposed to a single signature). It operates equivalently to the [`Signers` field](https://xrpl.org/docs/references/protocol/transactions/common-fields/#signers-field) used in standard transaction multi-sign. This field holds the signatures for the `Flags` and `TxnIDs` fields.
+
+### 2.6. Metadata
+
+The inner transactions will be committed separately to the ledger and will therefore have separate metadata. This is to ensure better backwards compatibility for legacy systems, so they can support `Batch` transactions without needing any changes to their systems.
+
+For example, a ledger that only has one `Batch` transaction containing 2 inner transactions would look like this:
+```
+[
+  OuterTransaction,
+  InnerTransaction1,
+  InnerTransaction2
+]
+```
+
+#### 2.6.1. Outer Transactions
+
+Each outer transaction will only contain the metadata for its sequence and fee processing, not for the inner transaction processing.
+
+There will also be a list of which transactions were actually processed, which is useful for the `ONLYONE` and `UNTILFAILURE` modes, since those may only process a subset of transactions, and for debugging with all modes. This section will be called `BatchExecutions`.
+
+It will contain a list of objects that have the following fields for every transaction that is processed (successfully or unsuccessfully):
+
+|FieldName | Required? | JSON Type | Internal Type |
+|:---------|:-----------|:---------------|:------------|
+|`TransactionHash`|✔️|`string`|`STUInt256`|
+|`InnerResult`|✔️|`string`|`STBlob`|
+
+Some important things to note:
+* It is possible that all transactions will not be included in this list. For example, when using the `ONLYONE` mode, if the first transaction succeeds, then the rest of the transactions will not even be processed. 
+* Transactions will only be included in the ledger if their result code is `tesSUCCESS` _and_ if the outer transaction has a result code of `tesSUCCESS`. For example, if the `ALLORNOTHING` mode is used and _any_ inner transaction fails, _none_ of the inner transactions will be included in the ledger.
+
+#### 2.6.2. Inner Transactions
+
+Each inner transaction will contain the metadata for its own processing. Only the inner transactions that were actually committed to the ledger will be included. This makes it easier for legacy systems to still be able to process `Batch` transactions as if they were normal.
+
+There will also be a pointer back to the parent outer transaction (`parent_batch`), for ease of development (similar to the `nftoken_id` field).
+
+## 3. Transaction Common Fields
+
+This standard doesn't add any new field to the [transaction common fields](https://xrpl.org/docs/references/protocol/transactions/common-fields/), but it does add another global transaction flag:
+
+| Flag Name | Value |
+|-----------|-------|
+|`tfInnerBatchTxn`|`0x40000000`|
```

Comment:
What is the necessity of the `tfInnerBatchTxn` flag? The existence of the `parent_batch` field in the meta-data already indicates that the current txn is an inner transaction. Furthermore, the field `Fee == 0` also confirms the usage of Batch transaction.

What is the intended use of this flag?

Reply:
`Fee = 0` does not confirm the usage of `Batch`. Anyone can submit a transaction with a fee of 0.

Reply:
Such trivial transactions will be rejected by rippled due to the load-based transaction cost factor (except for `KeyReset` transactions)

Reply:
> Such trivial transactions will be rejected by rippled

And how does rippled tell the difference between one of those trivial transactions and a `Batch` transaction?

Reply:
okay, that makes sense. You're trying to disambiguate an inner-`Batch` transaction from a non-`Batch` usual transaction. hmm okay